### ISSUE TITLE
Where should i modify if i wanna see learning rate progress with tensorboard(or something the other visual plots)

### ISSUE THREAD

**Issue by yellowjs0304** [ultralytics/ultralytics#4843]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, Thanks for your work.  

I wanna see the learning rate score progress, in the training step with tensorboard or the other visual plots also..  
Is there any option or code line that I can modify?

**Comment by yellowjs0304** [comment]
Thanks for the fast reply.
Then where should I modify it? Could you please share the part that you mentioned?
I can't find how can I put the "add_scalar(lr)" on the training code.

it looks like related to this [function](https://github.com/ultralytics/ultralytics/blob/e73447effb4262e171e79327c90a3cd2b91f751d/ultralytics/utils/callbacks/tensorboard.py#L18)

**Comment by glenn-jocher** [comment]
Hi @yellowjs0304,

Yes, you're on the correct path. The function you've pointed to is a part of the Tensorboard callback, which is responsible for logging various metrics. 

So to include the learning rate in the logs, you would need to augment this callback. Currently, it logs losses and some other parameters, but not the learning rate.

Locate the section of the code where the learning rate is updated (this is typically inside the training loop and often just after computing the gradients and before the optimizer's step method is called). Here, fetch the current learning rate from the optimizer (it's generally available under optimizer.param_groups[0]['lr']) and then use this value with the logger to include it in the Tensorboard log.

To make this change, you would have to understand how the callback system works in the Ultralytics' training loop, and carefully implement the changes without affecting other parts of the training.

If you're not completely confident making these changes, it might be safer to plot the learning rate externally using the output logs, which include the learning rate at every step.

Remember to test your changes thoroughly and look out for any unintended side effects. Hope this helps!

Cheers!
